### PR TITLE
remote tool: Log errors

### PR DESCRIPTION
**Pull request by vipersils**

### CODE REVIEW

File: packages/agent/src/tool/executor/toolPlugin.ts [hunk 1]
```diff
@@ -26,6 +25,15 @@ export function createToolPlugin({
 
           reply.status(200).send(textOutput);
         } catch (error: any) {
+          const logger = pino({
+            level: "debug",
+            messageKey: "message",
+          });
```
Comment (lgrammel) [inline packages/agent/src/tool/executor/toolPlugin.ts:31]: Can you make logger an additional argument of `createToolPlugin` and initialize it in the main server?

**Comment by vipersils** [comment]
this should be ready to go, do note that I'm usually working with atomic commits (i.e. each commit is an atomic change that will ideally pass build and be logically sane), do let me know if you need this separated into different PRs.

File: packages/agent/src/step/Step.ts [hunk 2]
```diff
@@ -38,12 +38,21 @@ export abstract class Step {
...
-    const result = await this._execute();
-    this.state = result;
-
-    this.run.onStepExecutionFinished({ step: this, result });
-
-    return result;
+    try {
+      const result = await this._execute();
+      this.state = result;
+      this.run.onStepExecutionFinished({ step: this, result });
```
Comment (lgrammel) [inline packages/agent/src/step/Step.ts:44]: `onStepExecutionFinished` should also be called in case of an error.

**Review by lgrammel — APPROVED** [review]
Atomic commits are great - no need to create a separate PR at this stage of the project.

File: packages/agent/src/step/Step.ts (hunk 2 above)
Comment (vipersils) [reply packages/agent/src/step/Step.ts]: done: https://github.com/lgrammel/js-agent/compare/cd7f250f16f97ddc7bbc56db3b7d12578376241f..804b8012f35b5ab6859d73d41d33c400940ecfa1

**Pull request merged**